x/participation/keeper/msg_withdraw_allocations.go

```diff
@@ -12,18 +12,21 @@ import (
 
 func (k msgServer) WithdrawAllocations(goCtx context.Context, msg *types.MsgWithdrawAllocations) (*types.MsgWithdrawAllocationsResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	blockTime := ctx.BlockTime()
 
 	auction, found := k.fundraisingKeeper.GetAuction(ctx, msg.AuctionID)
 	if !found {
 		return nil, sdkerrors.Wrapf(types.ErrAuctionNotFound, "auction %d not found", msg.AuctionID)
 	}
 
-	blockTime := ctx.BlockTime()
 	if !auction.IsAuctionStarted(blockTime) {
 		return nil, sdkerrors.Wrapf(types.ErrAllocationsLocked, "auction %d not yet started", msg.AuctionID)
 	}
 
-	// TODO check delay is reached
+	withdrawalDelay := k.WithdrawalDelay(ctx)
```

Comment:
I'm thinking would `!auction.IsAuctionStarted(blockTime)` actually be necessary?
`!auction.IsAuctionStarted(blockTime)` implies `!blockTime.After(auction.GetStartTime().Add(withdrawalDelay))`

Reply:
Yeah I agree.  As long as we enforce that the param is positive, then the first part is always implied

Reply:
> Looks good, I will take the time to do some further experimentation later in the day👍
> 
> Is this PR blocking for anything?

Would be nice if merged before I'm done with #648, but I need to work more on that before it might become blocking, so I would say no for now.